app/transport/rest/user.go

```diff
@@ -60,3 +62,36 @@ func (uh UserHandler) CreateUser() http.Handler {
 		uh.logger.Debugw("User successfully created", "ID", id)
 	})
 }
+
+// GetUserByID will handle user search
+func (uh UserHandler) GetUserByID() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		var user entities.PublicUser
+		id := mux.Vars(req)["id"]
+
+		if err := validateID(id); err != nil {
+			uh.logger.Warnf("Invalid ID in the request: %s", id)
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+
+			return
+		}
+
+		user, err := uh.usecase.GetUser(id)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
```

Comment:
Check by other error. Must not know about SQL here.

Reply:
+